test/jdk/sun/security/pkcs11/KDF/TestHKDF.java

```diff
@@ -610,6 +610,22 @@ private static void test_HKDF_after_ECDH_HkdfSHA256() throws Exception {
                 "6e09");
     }
 
+    private static void test_invalid_AES_key_size() {
+        printTestHeader("Test derivation of an invalid AES key size");
+        try {
+            KDF k = KDF.getInstance("HKDF-SHA256", p11Provider);
+            k.deriveKey("AES", HKDFParameterSpec.ofExtract()
+                    .thenExpand(null, 31));
+            throw new Exception("No exception thrown.");
```

Comment:
nit: "Expected InvalidAlgorithmParameterException not thrown" is clearer?

Reply:
This exception will be the cause, the wrapper exception will inform that `InvalidAlgorithmParameterException` was expected to be thrown.